### PR TITLE
Revert "Prevent WhitelistingTextInputFormatter to return a empty string if the current value does not satisfy the formatter (#57264)"

### PR DESCRIPTION
**Pull request by jason-simmons**
This reverts commit af9a409eda703cc5b2643c7e1944b2ca7e93621a.

The documentation for WhitelistingTextInputFormatter says that it filters
the text to the set of characters matched by the provided regex.
It is intended for use with a regex that describes a character class
(such as WhitelistingTextInputFormatter.digitsOnly, which matches on
RegExp(r'\d+')).
It is not designed to handle regexes that match an entire string
(such as the example seen in https://github.com/flutter/flutter/issues/54417).

The reverted commit will cause unexpected moves of the cursor position
in text fields that expect the old behavior.  This can be seen in
the phone number field of the text fields demo in Flutter Gallery.

### CODE REVIEW

**Review by justinmc — APPROVED** [review]
LGTM

Good catch!  I didn't realize that when I reviewed the PR.

**Comment by Piinks** [comment]
The bot won't merge this until the merge conflict is resolved.

**Pull request merged**
